core/templates/dev/head/i18n.js

```diff
@@ -0,0 +1,96 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Translation functions for Oppia.
+ *
+ * @author milagro.teruel@gmail.com (Milagro Teruel)
+ */
+
+// Translations of strings that are loaded in the front page. They are listed
+// here to be loaded synchronously with the script to prevent a FOUC or
+// Flash of Untranslated Content.
+// See http://angular-translate.github.io/docs/#/guide/12_asynchronous-loading
+var defaultTranslations = {
+  I18N_GALLERY_PAGE_TITLE: 'Gallery',
+  I18N_GALLERY_PAGE_SUBTITLE: 'Oppia',
+  I18N_GALLERY_LOADING: 'Loading',
+  I18N_SIGNUP_PAGE_SUBTITLE: 'Registration',
+  I18N_SIGNUP_PAGE_TITLE: 'Oppia',
+  I18N_GALLERY_SEARCH_PLACEHOLDER: 'What are you curious about?',
+  I18N_GALLERY_ALL_LANGUAGES: 'All Languages',
+  I18N_GALLERY_LANGUAGES_EN: 'English',
+  I18N_GALLERY_ALL_CATEGORIES: 'All Categories',
+  I18N_GALLERY_CREATE_EXPLORATION: 'Create exploration',
+  I18N_SIDEBAR_HOME_LINK: 'Home',
+  I18N_SIDEBAR_HOME_ABOUT: 'About',
+  I18N_SIDEBAR_PARTICIPATION_PLAYBOOK: 'Participation Playbook',
+  I18N_SIDEBAR_FORUM: 'Forum',
+  I18N_SIDEBAR_FOLLOW_US: 'Follow Us',
+  I18N_SIDEBAR_ADDITIONAL_LINK_SITE_FEEDBACK: 'Site Feedback',
+  I18N_TOPNAV_SIGN_IN: 'Sign in',
+  I18N_SIGNUP_REGISTRATION: 'Registration',
+  I18N_SIGNUP_LOADING: 'Loading'
+};
+
+oppia.constant('SUPPORTED_LANGUAGES', {
+  en: 'English',
+  es: 'Español'
+});
+
+oppia.controller('I18nFooter', [
+    '$rootScope', '$scope', '$translate', 'SUPPORTED_LANGUAGES',
+    function($rootScope, $scope, $translate, SUPPORTED_LANGUAGES) {
+  $scope.SUPPORTED_LANGUAGES = SUPPORTED_LANGUAGES;
+  // Changes the language of the translations.
+  $scope.changeLanguage = function(langCode) {
+    $translate.use(langCode);
+  };
+  // After loading default translations, change the language for the stored
+  // language if necessary.
+  $rootScope.$on('$translateLoadingSuccess', function() {
+    var currentLang = $translate.proposedLanguage() || $translate.use();
+    $translate.use(currentLang);
+  });
+}]);
+
+oppia.config([
+    '$translateProvider', 'SUPPORTED_LANGUAGES',
+    function($translateProvider, SUPPORTED_LANGUAGES) {
+  var availableLanguageKeys = [];
+  var availableLanguageKeysMap = {};
+  for (var prop in SUPPORTED_LANGUAGES) {
+    availableLanguageKeys.push(prop);
+    availableLanguageKeysMap[prop + '*'] = prop;
+  };
+  availableLanguageKeysMap['*'] = 'en';
+  $translateProvider.registerAvailableLanguageKeys(
+    availableLanguageKeys, availableLanguageKeysMap);
+
+  $translateProvider.translations('en', defaultTranslations);
+  $translateProvider.fallbackLanguage('en');
+
+  $translateProvider.determinePreferredLanguage();
+  $translateProvider.useCookieStorage();
+
+  $translateProvider.useStaticFilesLoader({
+    prefix: '/i18n/locale-',
+    suffix: '.json'
+  });
+  $translateProvider.preferredLanguage('en');
```

Comment:
This lines are the result of exhaustive try and error. It started as the documentation, and then trying to correct the FoUC I arrived to this strange result.
Line 91 can be deleted. It does not affect anything, I did some experiments today. What this line should be doing is to set the language to English for the default translations to be loaded, until the preferred language is resolved. When Spanish was selected I remember seen first no translations at all, then the English translations for a second and then finally the Spanish strings. Now, I never see the English translations. It may be the result of updating the libraries.

The chaining definitely makes it more easy to read. But doesn't it breaks the code style?